### PR TITLE
validate middleware

### LINKED ISSUE #75: validate middleware
compare results from multiple upstreams to ensure they are the same. useful to debug bad subway rpc

### PR DESCRIPTION
**Pull request by ermalkaleci**
closes #75

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,4 +75,18 @@ It's also possible to run individual benchmarks by:
...
+This is useful for debugging to make sure the returned values are as expected.
+You can use this by compiling with `validate` feature and enable validate middleware on your config file.
+```yml
+middlewares:
+  methods:
+    - validate
+```
+NOTE: Keep in mind that if you place `validate` middleware before `inject_params` you may get false positive errors because the request will not be the same.
+
+Use `.validateingore` file to list all the methods you want to ignore.
```
Comment (xlc) [inline README.md:92]: can we just configure it in the config.yml file?

Comment (ermalkaleci) [reply README.md:92]: Yes I did that but it will require to maintain 2 extensions list (one featured) and more code. This implementation is simple

Comment (xlc) [reply README.md:92]: but having two config file is just confusing

File: src/extensions/validator/mod.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        Self { config }
+    }
+
+    pub fn ignore(&self, method: &String) -> bool {
+        self.config.ignore_methods.contains(method)
+    }
+
+    pub fn validate(&self, client: Arc<Client>, request: CallRequest, response: CallResult) {
+        tokio::spawn(async move {
+            let healthy_endpoints = client.endpoints().iter().filter(|x| x.health().score() > 0);
```
Comment (xlc) [inline src/extensions/validator/mod.rs:40]: should we have a helper method `healthy_endpoints`?

File: src/extensions/validator/mod.rs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            futures::future::join_all(healthy_endpoints.map(|endpoint| async {
```
Comment (xlc) [inline src/extensions/validator/mod.rs:41]: do we need to filter out the current endpoint?

Comment (ermalkaleci) [reply src/extensions/validator/mod.rs:41]: response may come from cache, so why not hit them all

File: src/extensions/validator/mod.rs (hunk 2 above)
Comment (ermalkaleci) [reply src/extensions/validator/mod.rs:40]: yeah makes sense

**Review by xlc — APPROVED** [review]

**Comment by xlc** [comment]
maybe have an example config file

**Comment by ermalkaleci** [comment]
I think it's easy to copy/paste config from README and make use of validator

**Pull request merged**